wicket-core/src/main/java/org/apache/wicket/protocol/http/CsrfPreventionRequestCycleListener.java

```diff
@@ -345,13 +359,24 @@ public void onRequestHandlerResolved(RequestCycle cycle, IRequestHandler handler
 			IRequestablePage targetedPage = prh.getPage();
 			HttpServletRequest containerRequest = (HttpServletRequest)cycle.getRequest()
 				.getContainerRequest();
-			String sourceUri = getSourceUri(containerRequest);
+            String sourceUri = getSourceUri(containerRequest);
 
 			// Check if the page should be CSRF protected
 			if (isChecked(targetedPage))
 			{
-				// if so check the Origin HTTP header
-				checkRequest(containerRequest, sourceUri, targetedPage);
+				//check sec-fetch-site header and call the fetch metadata check if present
+				if (hasFetchMetadataHeaders(containerRequest))
+				{
+					WebResponse webResponse = (WebResponse)cycle.getResponse();
+					//TODO: should we define "Vary" as a header in ResourceIsolationPolicy / somewhere else?
+					webResponse.setHeader("Vary", "Sec-Fetch-Dest, Sec-Fetch-Site, Sect-Fetch-Mode");
```

Comment:
there's a typo in Sec-Fetch-Mode here. Let's write a test that verifies that these headers are set to the correct value and let's use constants like we do in the default resource isolation policy, please.

Reply:
+1 on defining the Vary keyword as a constant associated with ResourceIsolationPolicy